ibis/expr/operations.py

```diff
@@ -1862,22 +1885,27 @@ def root_tables(self):
     def can_add_filters(self, wrapped_expr, predicates):
         pass
 
-    def is_ancestor(self, other):
-        import ibis.expr.lineage as lin
-
-        if isinstance(other, ir.Expr):
-            other = other.op()
+    @staticmethod
+    def compare_argument_sequences(lefts, rights):
+        return not lefts or not rights or all_equal(lefts, rights)
 
+    def compatible_with(self, other):
+        # self and other are equivalent except for predicates, selections, or
+        # sort keys any of which is allowed to be empty. If both are not empty
+        # then they must be equal
         if self.equals(other):
             return True
 
-        expr = self.to_expr()
-        fn = lambda e: (lin.proceed, e.op())  # noqa: E731
-        for child in lin.traverse(fn, expr):
-            if child.equals(other):
-                return True
+        if not isinstance(other, type(self)):
+            return False
 
-        return False
+        return self.table.equals(other.table) and (
```

Comment:
The naming is a bit twisted, perhaps `empty_or_equal`?

Reply:
Sounds good, fixing ...